engines/private/private.cpp

```diff
@@ -737,6 +750,158 @@ void PrivateEngine::selectMask(Common::Point mousePos) {
 	}
 }
 
+bool PrivateEngine::selectLocation(Common::Point mousePos) {
```

Comment:
```suggestion
bool PrivateEngine::selectLocation(Common::Point &mousePos) {
```

And do the same on the methods below. This will avoid unnecessary object copying.